tests/hikari/internal/test_reflect.py

```diff
@@ -19,12 +19,18 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+import inspect
+import sys
 import typing
 
+import mock
+import pytest
+
 from hikari.internal import reflect
 
 
-class TestResolveSignature:
+@pytest.mark.skipif(sys.version_info >= (3, 10), reason="This strategy is specific to 3.10 > versions")
```

Comment:
Random point, but it just came to mind. Do we want to allow for specific version tests or do we want to allow all tests pass in any version of Python by mocking?

Reply:
Since the tests are run against every supported python ver in the pipelines this approach still gives full test coverage while being easier to implement and less likely to lead to stuff going unnoticed due to excessive mocking